src/IdentityServer4.WsFederation/WsFederationReturnUrlParser.cs

```diff
@@ -38,7 +36,7 @@ public bool IsValidReturnUrl(string returnUrl)
         {
             if (returnUrl.IsLocalUrl())
             {
-                var message = GetSignInRequestMessage(returnUrl);
+                var message = WsFederationMessageParser.GetSignInRequestMessage(returnUrl);
```

Comment:
I like the idea of moving it to a separate helper class and also that you add unit tests <3